### PR TITLE
configure lr scheduler

### PR DESCRIPTION
**Pull request by ncfrey**

### CODE REVIEW

File: src/lobster/model/_cbmlm.py [hunk 1]
```diff
@@ -291,11 +291,7 @@ def configure_optimizers(self):
         optimizer = torch.optim.AdamW(
             self.model.parameters(), lr=self._lr, betas=(self._beta1, self._beta2), eps=self._eps
         )
-        scheduler = get_linear_schedule_with_warmup(
-            optimizer,
-            num_warmup_steps=self._num_warmup_steps,
-            num_training_steps=self._num_training_steps,
-        )
+        scheduler = instantiate(self.scheduler_cfg, optimizer=optimizer)
```
Comment (karinazad) [inline src/lobster/model/_cbmlm.py:294]: where is `self.scheduler_cfg` set?

File: src/lobster/hydra_config/lr_scheduler/default.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+defaults:
+  - _self_
+
+scheduler:
```
Comment (karinazad) [inline src/lobster/hydra_config/lr_scheduler/default.yaml:4]: does it need to be _partial_?

File: src/lobster/model/modern_bert/_modern_bert.py [hunk 3]
```diff
@@ -41,7 +43,12 @@ def __init__(
         num_warmup_steps: int = 1_000,
         mask_percentage: float = 0.25,
         max_length: int = 512,
-        **model_kwargs,
+        scheduler: Literal["linear", "cosine", "cosine_with_restarts", "polynomial", "constant", "constant_with_warmup",
```
Comment (karinazad) [inline src/lobster/model/modern_bert/_modern_bert.py:46]: nice!

File: src/lobster/model/modern_bert/_modern_bert.py [hunk 4]
```diff
@@ -114,11 +124,20 @@ def configure_optimizers(self):
...
-        # TODO: Make this configurable
-        scheduler = get_linear_schedule_with_warmup(
+        # Create base kwargs for the scheduler
+        scheduler_params = {
+            "num_warmup_steps": self._num_warmup_steps,
+            "num_training_steps": self._num_training_steps,
+        }
+    
+        # Add any additional scheduler kwargs from initialization
+        if hasattr(self, 'scheduler_kwargs'):
```
Comment (karinazad) [inline src/lobster/model/modern_bert/_modern_bert.py:134]: isn't this always true?

File: tests/lobster/model/modern_bert/test__modern_bert.py [hunk 5]
```diff
@@ -12,14 +13,66 @@
...
+
+    def test_hydra_instantiate(self):
+        if not _FLASH_ATTN_AVAILABLE:
+            import pytest
+
+            pytest.skip("flash_attn not available")
+
+        # Define path to the config file
+        config_path = os.path.join(
+            os.path.dirname(os.path.dirname(os.path.dirname(os.path.dirname(os.path.dirname(__file__))))),
```
Comment (karinazad) [inline tests/lobster/model/modern_bert/test__modern_bert.py:36]: nit: could be `config_path = Path(__file__).parents[4]`

**Review by karinazad — APPROVED** [review]
